### PR TITLE
Allow configurable auth0

### PR DESCRIPTION
**Pull request by lorentzlasson**
https://github.com/stiftelsen-effekt/admin/issues/20

---

Tested on devices 
- [ ] Desktop 💻
- [ ] Mobile 📱

Tests  
- [ ] All tests are running ✔️
- [ ] Test are updated 🧪
- [ ] Code Review 👩‍💻
- [ ] QA 👌

⏲️ Time spent on CR:

⏲️ Time spent on QA:

### CODE REVIEW

File: src/config/config.ts [hunk 1]
```diff
@@ -1,16 +1,7 @@
-export const DEV_ENVIRONMENT: boolean = process.env.NODE_ENV === 'development' ? true : false;
-export const API_URL: string = process.env.REACT_APP_API_URL || 'http://localhost:5050';
-export const API_AUTH: IApiAuth = {
```
Comment (lorentzlasson) [inline src/config/config.ts:3]: Unused/old AFAICT

File: clouddeploy.dev.yml [hunk 2]
```diff
@@ -29,7 +35,13 @@ steps:
...
+  env:
+    - 'CI=true'
+    - 'REACT_APP_API_URL=https://api.dev.geeffektivt.se'
+    - 'REACT_APP_AUTH_DOMAIN=geeffektivt.eu.auth0.com'
+    - 'REACT_APP_AUTH_CLIENT_ID=LFS9efinrKEt1d6AbpVMjqHVIszj6TOv'
+    - 'REACT_APP_AUTH_AUDIENCE=geeffektivt.se'
+    - 'REACT_APP_AUTH_SCOPE="openid profile email read:donations write:donations read:profile write:profile read:distributions write:distributions read:agreements write:agreements admin"'
   args: ['run', 'build']
 
 - id: 'deploy-sweden'
```
Comment (lorentzlasson) [inline clouddeploy.dev.yml:47]: Example run https://console.cloud.google.com/cloud-build/builds;region=global/4246e79c-d96b-447f-9e64-6ee75d773700?authuser=1&project=hidden-display-243419

File: src/App.tsx [hunk 3]
```diff
@@ -10,20 +10,28 @@ import 'react-datepicker/dist/react-datepicker.css';
...
     <Provider store={store}>
       <Auth0Provider
-        domain={'gieffektivt.eu.auth0.com'}
-        clientId={'jg0jOryc8mhJ0Z0FVLfrDxx5qaznd2x8'}
-        audience="https://data.gieffektivt.no"
-        scope="openid profile email read:donations write:donations read:profile write:profile read:distributions write:distributions read:agreements write:agreements read:vipps_api write:vipps_api admin"
+        domain={AUTH_DOMAIN}
+        clientId={AUTH_CLIENT_ID}
+        audience={AUTH_AUDIENCE}
+        scope="admin"
```
Comment (lorentzlasson) [inline src/App.tsx:30]: Turns out this is the only one that's needed. `openid profile email` gets added automatically and the rest custom permissions are covered by the admin role

Example :point_down: 
<details>

```json
{
  "https://gieffektivt.no/user-id": 27,
  "https://gieffektivt.no/roles": [
    "admin",
    "donor"
  ],
  "iss": "https://gieffektivt.eu.auth0.com/",
  "sub": "auth0|63ff5798be60c526f4d3ac3a",
  "aud": [
    "https://data.gieffektivt.no",
    "https://gieffektivt.eu.auth0.com/userinfo"
  ],
  "iat": 1683296164,
  "exp": 1683382564,
  "azp": "jg0jOryc8mhJ0Z0FVLfrDxx5qaznd2x8",
  "scope": "openid profile email admin",
  "permissions": [
    "admin",
    "read:agreements",
    "read:distributions",
    "read:donations",
    "read:profile",
    "read:vipps_api",
    "write:agreements",
    "write:distributions",
    "write:donations",
    "write:profile",
    "write:vipps_api"
  ]
}
```

**Review by fellmirr — APPROVED** [review]
Perfect 👍

**Pull request merged**